crates/autopilot/src/domain/settlement/mod.rs

```diff
@@ -176,3 +221,285 @@ impl From<infra::persistence::DatabaseError> for Error {
         Self::Infra(err.0)
     }
 }
+
+#[cfg(test)]
+mod tests {
```

Comment:
Did you add a test case here for a transaction that settled CoW AMM?

Reply:
no, will add one

Reply:
added `settlement_with_cow_amm`